src/robot_state_publisher.cpp

```diff
@@ -129,10 +129,15 @@ RobotStatePublisher::RobotStatePublisher(const rclcpp::NodeOptions & options)
   tf_broadcaster_ = std::make_unique<tf2_ros::TransformBroadcaster>(this);
   static_tf_broadcaster_ = std::make_unique<tf2_ros::StaticTransformBroadcaster>(this);
 
+  auto publisher_options = rclcpp::PublisherOptions();
+  // Explicitly disable intra_process_comms for this publisher alone to allow
+  // for this node to run in a component container with intra_process_comms enabled.
+  publisher_options.use_intra_process_comm = rclcpp::IntraProcessSetting::Disable;
```

Comment:
I'm surprised this is necessary.  Does this mean subscribers in the same process don't receive messages from publishers if the publishers have transient_local durability? I would have expected them to still get the messages through the rmw API.

Reply:
In Iron and before an exception would be thrown by rclcpp. I wasn't aware that support for this was added in Jazzy/Rolling. So we can close this PR